### DISCUSSION TITLE
Issues with composer when installing Open CSP

### DISCUSSION THREAD

**Discussion by dafogary** [Open-CSP/open-csp#14] [Q&A]
Hi

Apologies if this isn't where I should be getting support, I couldn't find any information on where I needed to ask for support from.

I am trying to install open-csp as a proof of concept, but I have run into an error.

I install using:

sudo -u www-data ./install_open_csp.sh /var/www/html/opencsp

```
Step 2: Running composer.
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Symfony/Component/Console/Command/DumpCompletionCommand.php:48
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Symfony/Component/Console/Command/DumpCompletionCommand.php:56
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Composer/Autoload/AutoloadGenerator.php:879
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Composer/Autoload/AutoloadGenerator.php:884
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Symfony/Component/Console/Command/DumpCompletionCommand.php:48
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Symfony/Component/Console/Command/DumpCompletionCommand.php:56
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Composer/Autoload/AutoloadGenerator.php:879
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Composer/Autoload/AutoloadGenerator.php:884
ComposerHookHandler::onPreUpdate
Loading composer repositories with package information
Updating dependencies
Deprecation Notice: Using ${var} in strings is deprecated, use {$var} instead in /usr/share/php/Composer/DependencyResolver/Problem.php:366
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires open-csp/wiki-search v4.3.0 -> satisfiable by open-csp/wiki-search[v4.3.0].
    - open-csp/wiki-search v4.3.0 requires php ^7.4 -> your php version (8.3.12) does not satisfy that requirement.
  Problem 2
    - Root composer.json requires wikibase-solutions/flex-form ^1.1 -> satisfiable by wikibase-solutions/flex-form[v1.1.0, ..., v1.1.45].
    - wikibase-solutions/flex-form[v1.1.0, ..., v1.1.45] require php ^7.3.19 -> your php version (8.3.12) does not satisfy that requirement.
  Problem 3
    - Root composer.json requires wikibase-solutions/page-sync ^2.1 -> satisfiable by wikibase-solutions/page-sync[v2.1.0, ..., v2.6.5].
    - wikibase-solutions/page-sync[v2.1.0, ..., v2.6.5] require php ^7.3.19 -> your php version (8.3.12) does not satisfy that requirement.
  Problem 4
    - onoi/tesa 0.1.0 requires wikimedia/cdb ~1.0 -> found wikimedia/cdb[1.0.0, ..., 1.4.1] but it conflicts with your root composer.json require (2.0.0).
    - mediawiki/semantic-media-wiki 3.2.3 requires onoi/tesa ~0.1 -> satisfiable by onoi/tesa[0.1.0].
    - Root composer.json requires mediawiki/semantic-media-wiki 3.2.3 -> satisfiable by mediawiki/semantic-media-wiki[3.2.3].

Running update with --no-dev does not mean require-dev is ignored, it just means the packages will not be installed. If dev requirements are blocking the update you have to resolve those problems.


The installation was unsuccessfull.
Please read the error messages above, and act accordingly.
```
What do I need to do to resolve this?

**Comment by LarsS88** [comment]
Hello @dafogary,

From your composer output it seems apparent that you've cloned the master branch of the project. This branch contains version 1 of Open-CSP, which is not compatible with PHP 8.x.
In order to have any chance at all of installing in a PHP 8.3 environment you'll need to use the REL1_39 branch of the project; Open-CSP 2.0, based on MediaWiki 1.39 LTS.

The requirements there are the following:
> PHP version >=8.0 < 9
> Composer version >=2
> Mediawiki version 1.39
> ElasticSearch version 6.8

I'd like to add also just as an FYI that I'm personally only aware of Open-CSP 2.0 having been tested and proven working with PHP 8.0 and 8.1.
So whilst it should probably be working with 8.3, there's no guarantee.
All installations of Open-CSP 2.0 we currently have running at Wikibase Solutions are using either 8.1.2 or 8.1.29.

**Reply by dafogary** [reply]
Thanks for the reply, apologies for the delay. I have been investigating more.

To allow me to install, I have had to downgrade to PHP 8.1, but I am now having issues with the ElasticSearch.

I have installed ElasticSearch 6.8.23, but get the error:

`Elasticsearch\Common\Exceptions\ServerErrorResponseException from line 703 of /var/www/html/opencsp/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php: {"error":{"root_cause":[{"type":"class_cast_exception","reason":"class java.util.ArrayList cannot be cast to class java.util.Map (java.util.ArrayList and java.util.Map are in module java.base of loader 'bootstrap')"}],"type":"class_cast_exception","reason":"class java.util.ArrayList cannot be cast to class java.util.Map (java.util.ArrayList and java.util.Map are in module java.base of loader 'bootstrap')"},"status":500}`

If I run curl -X GET http://localhost:9200

I get:

```
{
  "name" : "CA8nESe",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "4u94ug8qTPmTG2CdGJOXGA",
  "version" : {
    "number" : "6.8.23",
    "build_flavor" : "default",
    "build_type" : "deb",
    "build_hash" : "4f67856",
    "build_date" : "2022-01-06T21:30:50.087716Z",
    "build_snapshot" : false,
    "lucene_version" : "7.7.3",
    "minimum_wire_compatibility_version" : "5.6.0",
    "minimum_index_compatibility_version" : "5.0.0"
  },
  "tagline" : "You Know, for Search"
}
```
The Java I have is 17.0.11 2024-04-16 LTS, I have set /usr/lib/jvm/openjdk-17 as Java Home, but I did get an error before, but struggling to replicate it.

Any help is apprecited
